doc/docs.md

```diff
@@ -4105,12 +4105,35 @@ fn copy_all(dupes bool) {
 }
 ```
 
+It also possible to use multiline comments with `/* */` (they can also be mixed with single line comments)
+
+```v
+/*
+copy_all recursively copies all elements of the array by their value,
+if `dupes` is false all duplicate values are eliminated in the process.
+*/
+fn copy_all(dupes bool) {
+	// ...
+}
+```
+
 By convention it is preferred that comments are written in *present tense*.
 
 An overview of the module must be placed in the first comment right after the module's name.
 
 To generate documentation use vdoc, for example `v doc net.http`.
 
+### Newlines in Documentation Comments
+
+Comments spanning multiple lines are merged together using spaces, unless
+
+- the line is empty
+- the line ends with a `.` (end of sentence)
+- the line is contains purely of at least 3 of `-`, `=`, `_`, `*`, `~` (horizontal rule)
```

Comment:
From here down are markdown rules.  Is markdown format allowed in doc comments?  If so, markdown's "two or more spaces at the end of a line means a newline" rule needs to be added back.

If markdown format isn't supposed to be allowed, the rules need to be removed as well.

Reply:
Markdown format is allowed and used (think of `codeblocks`).

I've removed the 2 space rule for now, because vfmt removes trailing whitespaces and complained about my test case for it.

Once I find out how to patch that behavior out of vfmt (ofc limited to comments), I'll add it back (no idea about the inner workings of vfmt currently, so that might take a while).

Reply:
Markdown is supported only for .md files, not in ordinary comments.

Reply:
But... it is supported.  
If you look at the html docs, the comments were passed through markdown or links and `codeblocks` wouldn't work.

Also, why can't they be markdown?

I looked through the docs and didn't see any obvious issues.
It'd also allow for more formatting options (`**bold**`, `> quote` etc etc).

Reply:
It is used, but not supported, meaning that we should not cater to support every little markdown edge case like the spaces at end of lines etc.

Reply:
They are not supported and I don't plan on adding them back (vmft inner workings is a beast).